### ISSUE TITLE
Stored request not merged when using {{UUID}}

### ISSUE THREAD

**Issue by rsnigel** [prebid/prebid-server#2045]
When a stored request uses the new `{{UUID}}` macro or generateRequestId is enabled in the configuration, a stored top level request is not merged with the openrtb2 auction request.

Example stored request:
```
{
  "id": "{{UUID}},
  ...
}
```

Example auction request:
```
{
  "ext": {
    "prebid": {
      "storedrequest": {
        "id": "example"
      }
    }
  }
}
```

In this example, prebid-server loads the top level stored request `example` and in line https://github.com/prebid/prebid-server/blob/master/endpoints/openrtb2/auction.go#L1371 it checks if `{{UUID}}` is used.

`if deps.cfg.GenerateRequestID || bidRequestID == "{{UUID}}" {`

This causes that the stored request will never be merged, because the else part in line https://github.com/prebid/prebid-server/blob/master/endpoints/openrtb2/auction.go#L1392 will never be triggered.

Expected result:
{{UUID}} from stored request is replaced by a real UUID and then the stored request is merged with the openrtb2 auction request.

**Comment by SyntaxNode** [comment]
Gotcha. Thank you for expanded example. We'll look into it.

**Comment by rsnigel** [comment]
The issue is that I took over the example from @SyntaxNode , but this example works because an app.id is specified. The issue only happens in non-app mode.

Here a complete example.

Stored request with id `test-uuid`:
```
{
  "id": "{{UUID}}",
  "cur": [
    "USD"
  ]
}
```

Use Postman (or any other tool) to send the following openrtb2 request:
```
{
    "id": "some-request-id",
    "imp": [{
        "id": "some-impression-id",
        "banner": {
            "format": [{
                    "w": 600,
                    "h": 500
                },
                {
                    "w": 300,
                    "h": 600
                }
            ]
        },
        "ext": {
            "appnexus": {
                "placementId": 12883451
            }
        }
    }],
    "ext": {
        "prebid": {
            "debug": true,
            "storedrequest": {
                "id": "test-uuid"
            }
        }
    },
  "site": {
    "page": "https://example.com"
  }
}
```

Look at response `ext.debug.resolvedrequest`, which is missing the `currency` field:
```
{
  "id":"some-request-id",
  "imp":[
    {
      "id":"some-impression-id",
      "banner":{
        "format":[
          {
            "w":600,
            "h":500
          },
          {
            "w":300,
            "h":600
          }
        ]
      },
      "ext":{
        "appnexus":{
          "placementId":12883451
        }
      }
    }
  ],
  "site":{
    "page":"https://example.com",
    "ext":{
      "amp":0
    }
  },
  "device":{
    "ua":"PostmanRuntime/7.28.4",
    "ip":"81.109.219.23"
  },
  "at":1,
  "source":{
    "tid":"6f47fb9f-43aa-4b97-949f-218525f58ebe"
  },
  "ext":{
    "prebid":{
      "debug":true,
      "storedrequest":{
        "id":"test-uuid"
      }
    }
  }
}
```

Now either remove the `id` field from the stored request or change {{UUID}} to some other value, like this example:
```
{
  "id": "other-request-id",
  "cur": [
    "USD"
  ]
}
```

Send the same request again and look at response `ext.debug.resolvedrequest`, which now has a `currency` field available:
```
{
  "id":"some-request-id",
  "imp":[
    {
      "id":"some-impression-id",
      "banner":{
        "format":[
          {
            "w":600,
            "h":500
          },
          {
            "w":300,
            "h":600
          }
        ]
      },
      "ext":{
        "appnexus":{
          "placementId":12883451
        }
      }
    }
  ],
  "site":{
    "page":"https://example.com",
    "ext":{
      "amp":0
    }
  },
  "device":{
    "ua":"PostmanRuntime/7.28.4",
    "ip":"81.109.219.23"
  },
  "at":1,
  "cur":[
    "USD"
  ],
  "source":{
    "tid":"7d37d307-7c3c-48ba-a06d-6cc2134cb6ed"
  },
  "ext":{
    "prebid":{
      "debug":true,
      "storedrequest":{
        "id":"test-uuid"
      }
    }
  }
}
```

This complete example should make it now easy to reproduce. Tested on prebid-server v0.179.0.

**Comment by AlexBVolcy** [comment]
Just want to update this thread:

There are some new updates that Bret suggested to be made to the UUID Generation feature that was initially implemented in these PRs:

#1938

#2000

Another conversation will be had to Bret to confirm these updates, but essentially he wants the UUID to only be generated and updated for the request for Amp Auctions. The UUID should still be generated for other types of auctions, but not actually updated for the request.

**Comment by AlexBVolcy** [comment]
This bug was addressed in this PR https://github.com/prebid/prebid-server/pull/2203 and was released in PBS-Go v0.206.0
